client/src/node_api/core/mod.rs

```diff
@@ -75,4 +75,40 @@ impl Client {
         }
         Ok(outputs)
     }
+
+    pub async fn try_get_metadata_for_outputs(&self, output_ids: Vec<OutputId>) -> Result<Vec<OutputMetadataResponse>> {
+        let mut output_metadata_responses = Vec::new();
+
+        #[cfg(target_family = "wasm")]
+        for output_id in output_ids {
+            if let Ok(output_metadata_response) = self.get_output_metadata(&output_id).await {
+                output_metadata_responses.push(output_metadata_response);
+            }
+        }
+
+        #[cfg(not(target_family = "wasm"))]
+        for output_ids_chunk in output_ids.chunks(MAX_PARALLEL_API_REQUESTS).map(<[OutputId]>::to_vec) {
+            let mut tasks = Vec::new();
+            for output_id in output_ids_chunk {
+                let client_ = self.clone();
+
+                tasks.push(async move {
+                    tokio::spawn(async move {
+                        // Ignore possible errors
+                        if let Ok(output_metadata_response) = client_.get_output_metadata(&output_id).await {
+                            Some(output_metadata_response)
+                        } else {
+                            None
+                        }
```

Comment:
I think you can use `client_.get_output_metadata(&output_id).await.ok()` here, please also update in the other function if it works :trollface: